examples/rtctest/rtctest.c

```diff
@@ -1,13 +1,17 @@
 #include <stdio.h>
+#include <sys/time.h>
 #include <malloc.h>
-#include <string.h>
 #include <libdragon.h>
 
 #define BLACK 0x000000FF
 #define WHITE 0xFFFFFFFF
 
 #define JOYSTICK_DEAD_ZONE 32
 
+#define MAX(a,b)  ({ typeof(a) _a = a; typeof(b) _b = b; _a > _b ? _a : _b; })
+#define MIN(a,b)  ({ typeof(a) _a = a; typeof(b) _b = b; _a < _b ? _a : _b; })
+#define CLAMP(x, min, max) (MIN(MAX((x), (min)), (max)))
```

Comment:
Are these part of the C standard library yet?

Reply:
NOPE. I think they just got added to C23 for *floats*, but nah, nothing that I know of as a generic macro... I even have these in my own codebase.

Reply:
LibDragon has them in the "private header" `#include "utils.h"`: https://github.com/DragonMinded/libdragon/blob/8fcaba63faff7c73411606617044c295501fe936/src/utils.h#L12-L14

They seem useful enough that we should expose them to LibDragon devs, but they're generic enough that exposing them in `<libdragon.h>` could cause naming conflicts. (Basically every C code base has these peppered around in some form or another).

I know LibDragon strongly encourages including the entire library as a single header, but I definitely see the benefits for `utils.h` to be exposed as an "optional public header": `#include <libdragon/utils.h>` or something.

This is another one of those "it's a little late to change now" things, but sometimes I wish LibDragon used a common prefix so that we wouldn't have to worry (as much) about namespace collisions with other libraries and user code.

Reply:
A solution could be hiding them behind an opt-in preprocessor define. Like Microsoft does for defining `M_PI` and friends in `math.h`. I think you always get them with GCC but with Microsoft's implementation, you need `#define _USE_MATH_DEFINES` to have them defined.